File: connections/CreateWikiPage.js
/*
 * Antes de executar:
 *
 * > npm install request
 * > npm install xml2js
 *
 *
 */
'use strict';
// Ignore self signed certificates
process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';

const request = require('request');
const xml2js = require('xml2js');
const parser = new xml2js.Parser();

const CONNECTIONS_HOST = 'https://connections.<REPLACE_HERE>.com';
const CONNECTIONS_USERNAME = '<REPLACE_HERE>';
const CONNECTIONS_PASSWORD = '<REPLACE_HERE>';

// For Wikis inside a community -> https://{hostname}/wikis/home?lang=en-us#!/wiki/W34c618febb3c_4a55_81b7_0d5a81dc1954
// For Wikis outside a community -> https://{hostname}/wikis/home?lang=en-us#!/wiki/User Guide
var wiki_id_or_label = 'W34c618febb3c_4a55_81b7_0d5a81dc1954';

function createWikiPage(wiki_id_or_label) {

    var body = '';
    body += '<entry xmlns="http://www.w3.org/2005/Atom">';
    body += '<title type="text">My First Wiki</title>';
    body += '<content type="text/html">';
    body += '<![CDATA[<?xml version="1.0" encoding="UTF-8"?><p>';
    body += '<strong>Hello World!</strong>';
    body += '</p>]]>';
    body += '</content>';
    body += '<category term="wikipagetag1" />';
    body += '<category term="wikipagetag2" />';
    body += 'category scheme="tag:ibm.com,2006:td/type" term="page" label="page" />';
    body += '</entry>';

    var options = {
        url: CONNECTIONS_HOST + '/wikis/basic/api/wiki/' + wiki_id_or_label  + '/feed',
        auth: {
            username: CONNECTIONS_USERNAME,
            password: CONNECTIONS_PASSWORD
        },
        headers: {
            'Content-Type': 'application/atom+xml;charset=UTF-8'
        },
        body: body
    };

    request.post(options, function(error, response, body) {
        if (error) {
            console.error('error --> ');
            console.error(error);
        }
       if (response.statusCode >= 200 && response.statusCode < 300) {
            console.log('Success')
            console.error('response.statusCode --> ', response.statusCode);
            console.error('body --> ');
            console.error(body);
            console.error('\n');
        } else {
            console.log('Failed')
            console.error('response.statusCode --> ', response.statusCode);
            console.error('body --> ');
        }


     });
}


//******************************     MAIN          **********************************//
console.log('Connecting to IBM Connections...\n');
console.log('Creating Wiki Page...\n');
createWikiPage(wiki_id_or_label);

File: ibm-connections/CreateWikiPage.py
# -*- coding: utf-8 -*-
#
# Necessary libraries:
#
# > pip install requests
#
# For documentation on Tone Analyzer:
#    https://watson-api-explorer.mybluemix.net/apis/tone-analyzer-v3#!/tone/GetTone
#
import sys
import requests
from requests.packages.urllib3.exceptions import InsecureRequestWarning
from requests.auth import HTTPBasicAuth

requests.packages.urllib3.disable_warnings(InsecureRequestWarning)

CONNECTIONS_HOST = 'https://connections.<company>.com'
CONNECTIONS_USERNAME = '<REPLACE_HERE>'
CONNECTIONS_PASSWORD = '<REPLACE_HERE>'

wiki_id_or_label = 'W34c618febb3c_4a55_81b7_0d5a81dc1954';

xml_data = ''
xml_data += '<entry xmlns="http://www.w3.org/2005/Atom">'
xml_data += '<title type="text">My First Wiki Python</title>'
xml_data += '<content type="text/html">'
xml_data += '<![CDATA[<?xml version="1.0" encoding="UTF-8"?><p>'
xml_data += '<strong>Hello World!</strong>'
xml_data += '</p>]]>'
xml_data += '</content>'
xml_data += '<category term="wikipagetag1" />'
xml_data += '<category term="wikipagetag2" />'
xml_data += 'category scheme="tag:ibm.com,2006:td/type" term="page" label="page" />'
xml_data += '</entry>'

xml_data2 = '''
<entry xmlns="http://www.w3.org/2005/Atom">
<title type="text">My First Wiki Python 2</title>
<content type="text/html">
<![CDATA[<?xml version="1.0" encoding="UTF-8"?><p>
<strong>Hello World!</strong>
</p>]]>
</content>
<category term="wikipagetag1" />
<category term="wikipagetag2" />
category scheme="tag:ibm.com,2006:td/type" term="page" label="page" />
</entry>
'''

def createWikiPage(wiki_id_or_label, xml_data):

    headers = { 'Content-Type': 'application/atom+xml;charset=UTF-8'}

    url = CONNECTIONS_HOST + '/wikis/basic/api/wiki/' + wiki_id_or_label  + '/feed'
    auth=HTTPBasicAuth(CONNECTIONS_USERNAME, CONNECTIONS_PASSWORD)

    res = requests.post(url=url,headers=headers,auth=auth,verify=False,data=xml_data)
    if (res.status_code != 200):
        print 'doGetToneAnalyze: requests.post -> %s = %s\n' % (res.url, res)
        print res.content
        return None;

    return res.json()

#################### Main Module ###################
print 'Connecting to IBM Connections...\n'

print 'Creating Wiki Page...\n'
createWikiPage(wiki_id_or_label,xml_data)
createWikiPage(wiki_id_or_label,xml_data2)
